applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -339,6 +450,140 @@ static void uart_recovery(struct k_work *work)
 	LOG_DBG("UART recovered");
 }
 
+static void raw_send(struct k_work *work)
+{
+	int err;
+
+	ARG_UNUSED(work);
+
+	LOG_HEXDUMP_DBG(at_buf, at_buf_len, "RX");
+
+	if (slm_tcp_get_datamode()) {
+		(void)slm_tcp_send_datamode(at_buf, at_buf_len);
+		goto sent;
+	}
+
+	if (slm_udp_get_datamode()) {
+		(void)slm_udp_send_datamode(at_buf, at_buf_len);
+		goto sent;
+	}
+
+sent:
+	err = uart_rx_enable(uart_dev, uart_rx_buf[0],
+			     sizeof(uart_rx_buf[0]), UART_RX_TIMEOUT_MS);
+	if (err) {
+		LOG_ERR("UART RX failed: %d", err);
+	}
+	rx_start = k_uptime_get();
+	at_buf_len = 0;
+}
+
+static void inactivity_timer_handler(struct k_timer *timer)
+{
+	ARG_UNUSED(timer);
+
+	LOG_INF("time limit reached, size: %d", at_buf_len);
+
+	if (at_buf_len > 0) {
+		uart_rx_disable(uart_dev);
+		k_work_submit(&raw_send_work);
+	}
+}
+
+K_TIMER_DEFINE(inactivity_timer, inactivity_timer_handler, NULL);
+
+static void silence_timer_handler(struct k_timer *timer)
+{
+	ARG_UNUSED(timer);
+
+	/* Drop pending data unconditionally */
+	if (datamode_time_limit > 0) {
+		k_timer_stop(&inactivity_timer);
+	}
+	at_buf_len = 0;
+
+	/* quit datamode */
+	(void)exit_datamode();
+	if (slm_tcp_get_datamode()) {
+		slm_tcp_set_datamode_off();
+	}
+	if (slm_udp_get_datamode()) {
+		slm_udp_set_datamode_off();
+	}
+	datamode_off_pending = false;
+	/* send URC */
+	rsp_send(OK_STR, sizeof(OK_STR) - 1);
+}
+
+K_TIMER_DEFINE(silence_timer, silence_timer_handler, NULL);
+
+static int raw_rx_handler(const uint8_t *data, int datalen)
+{
+	const char *quit_str = CONFIG_SLM_DATAMODE_TERMINATOR;
+	int quit_str_len = strlen(quit_str);
+	int64_t silence_t = CONFIG_SLM_DATAMODE_SILENCE * MSEC_PER_SEC;
```

Comment:
Why `_t` suffix? It looks a bit odd on a variable.

Reply:
Removed